src/pages/[[...slug]].tsx

```diff
@@ -26,49 +30,68 @@ export default function ResourcePage({ resource, globalElements }) {
       <Head>
         <title>{title}</title>
       </Head>
-      {resource.type === 'node--news_story' && <NewsStory {...resource} />}
-      {resource.type === 'node--story_listing' && (
+      {resource.type === RESOURCE_TYPES.STORY_LISTING && (
         <StoryListing {...resource} />
       )}
-      {resource.type === 'node--q_a' && <QuestionAnswer {...resource} />}
+      {resource.type === RESOURCE_TYPES.STORY && <NewsStory {...resource} />}
+      {resource.type === RESOURCE_TYPES.QA && <QuestionAnswer {...resource} />}
     </Wrapper>
   )
 }
 
-export async function getStaticPaths(context): Promise<GetStaticPathsResult> {
+export async function getStaticPaths(
+  context: GetStaticPathsContext
+): Promise<GetStaticPathsResult> {
+  // `getStaticPaths` is run on every request in dev mode (`next dev`). We don't need this,
+  // so we set SSG=true on `next build/export` and SSG=false on `next dev`.
+  // `getStaticPaths` will never be called during runtime (`next start`), but we could set
+  // SSG=false there as well, for good measure.
+  if (!process.env.SSG) {
+    return {
+      paths: [],
+      fallback: 'blocking',
+    }
+  }
+
+  const storyListingPaths = await getAllStoryListingStaticPaths(context)
+  const storyPaths = await getStaticPathsByResourceType(
+    RESOURCE_TYPES.STORY,
+    context
+  )
+  const qaPaths = await getStaticPathsByResourceType(RESOURCE_TYPES.QA, context)
+
   return {
-    paths: await drupalClient.getStaticPathsFromContext(
-      Array.from(RESOURCE_TYPES),
-      context
-    ),
+    paths: [...storyListingPaths, ...storyPaths, ...qaPaths],
     fallback: 'blocking',
   }
 }
 
-export async function getStaticProps(context) {
-  const path = await drupalClient.translatePathFromContext(context)
+export async function getStaticProps(context: GetStaticPropsContext) {
+  const { slug = [] } = context.params
+  const [path, pageNumber] = getPathAndPageNumberFromSlug(slug)
 
-  if (!path) {
+  const pathInfo = await drupalClient.translatePath(path)
+  if (!pathInfo) {
     return {
       notFound: true,
     }
   }
 
-  const type = path.jsonapi.resourceName as (typeof RESOURCE_TYPES)[number]
-
-  if (!RESOURCE_TYPES.includes(type)) {
+  const resourceType = pathInfo.jsonapi
+    .resourceName as (typeof RESOURCE_TYPES)[keyof typeof RESOURCE_TYPES]
+  if (!Object.values(RESOURCE_TYPES).includes(resourceType)) {
     return {
       notFound: true,
     }
   }
 
-  const resource = await queries.getData(type, {
+  const resource = await queries.getData(resourceType, {
     context,
-    id: path.entity.uuid,
+    id: pathInfo?.entity?.uuid,
```

Comment:
why are these ? added here? if there's not pathinfo, we return up above. if we get to this point, it should have all the data we need, no?

Reply:
I tend to get overdefensive on some of this prop drilling, so this is probably partially out of habit. After all, there's nothing inherently _wrong_ with having the extra checks. It ultimately boils down to how we are trapping expected errors, etc. 

But I agree it is partially unnecessary here. We do know that `pathInfo` will be defined, so we don't need the first `?`. If we have `pathInfo.entity`, it will never result in a `ReferenceError`. I think that means it's likely _almost_ impossible for `entity` to not be defined, because we know what `pathInfo` _should_ look like, but I think we might be remiss to make that assumption. The goal here was to ensure that `undefined` would be passed rather than raising an uncaught error. As I said, I don't think it's "wrong" as it is, but I really appreciate your thorough review and attention to detail here, and I agree that this isn't the most pure way of handling this. My vote would be for removing the first `?` and keeping the second `?`. 

Note: This will have the result of possibly passing `undefined` to the `getData` function. We might need to look at that function to make sure it's handling bad data properly.

Reply:
Why would we ever want to pass `undefined` though? We're inside getStaticProps, we'll need the uuid every time. If it's a paginated page, we still want the uuid of the base listing page and then the correct page #

Reply:
It's not that we would _want_ to pass `undefined`. It's that we want to handle the possibility that it _is_ `undefined` ("it" is actually `pathInfo.entity`, to be clear).

### `pathInfo.entity?.uuid`
In this case, if `pathInfo.entity` is `undefined`, we set `id: undefined`. The call to `queries.getData` _should_ handle this possibility and return `undefined` as the resource.
### `pathInfo.entity.uuid`
In this case, if `pathInfo.entity` is undefined, an error is thrown and things break.